Documentation/devel/distribution.md

```diff
@@ -0,0 +1,145 @@
+
+### Current rkt distributions
+
+In rkt there are currently three kind of distribution: Appc, Archive and Docker.
+
+**Appc**
+This is a Name based distribution
+
+Appc defines a distribution using appc image discovery
+The format is: `cimd:appc:v=0:name?label01=....&label02=....`
+The scheme is "appc"
+the labels values must be Query escaped
+Example: `cimd:appc:v=0:coreos.com/etcd?version=v3.0.3&os=linux&arch=amd64`
+
+**ACIArchive**
+This is a Location based distribution
+
+ACIArchive defines a distribution using an archive file
+
+The format is: `cimd:aci-archive:v=0:ArchiveURL?query...`
+The distribution type is "aci-archive"
+ArchiveURL must be query escaped
+
+Examples:
+`cimd:aci-archive:v=0:file%3A%2F%2Fabsolute%2Fpath%2Fto%2Ffile`
+`cimd:aci-archive:v=0:https%3A%2F%2Fexample.com%2Fapp.aci`
+
+**Docker**
+This is a Name based distribution
+
+Docker defines a distribution using a docker registry
+The format is the same as the docker image string format (man docker-pull) with the "docker" scheme:
+`cimd:docker:v=0:[REGISTRY_HOST[:REGISTRY_PORT]/]NAME[:TAG|@DIGEST]`
+Examples:
+`cimd:docker:v=0:busybox`
+`cimd:docker:v=0:busybox:latest`
+`cimd:docker:v=0:registry-1.docker.io/library/busybox@sha256:a59906e33509d14c036c8678d687bd4eec81ed7c4b8ce907b888c607f6a1e0e6`
+
+### Future distributions
+
+**OCI Image distributions**
+This is a Name based distribution
+
+Now a docker distribution can also provide oci image but in future oci image will define one or more own kind of distribution starting from an image name (and additional tags, labels)
+
+**OCI Image layout**
+This is a Location based distribution
+
+It can fetch an image starting from a [OCI image layout](https://github.com/opencontainers/image-spec/blob/master/image-layout.md) format. The location can point to a single file archive, to a local/remote directory based layout or other kind of locations
+
+Probably it will be the sub-distribution used by the OCI image distribution described above) (like ACIArchive is the sub-distribution for the appc distribution).
+
+`cimd:oci-image-layout:v=0:file%3A%2F%2Fabsolute%2Fpath%2Fto%2Ffile?ref=refname`
+`cimd:oci-image-layout:v=0:https%3A%2F%2Fdir%2F?ref=refname`
+
+Since the OCI image layout can provide multiple images selectable by a ref there's the need to specify which ref to use in the archive distribution URI (see the above ref query parameter). Since distribution just covers one image it's not possible to import all the refs with just a distribution URI.
+
+TODO(sgotti). Define if oci-image-layout should internally handle both archive and directory based layouts or use two different distributions or a query parameter the explicitly define the layout (to avoid guessing if the url points to a single file or to a directory).
```

Comment:
My understanding it that one of the point for this is to stop guessing, so it may make sense to have different types if they need to be handled separately. But I think we should postpone this decision to the actual point when we need to reference those.